### PR TITLE
snap package.

### PR DESCRIPTION
**Pull request by mikeroyal**
Includes snapcraft.yaml to build snap.

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      last_committed_tag_ver="$(echo ${last_committed_tag} | sed 's/v//')"
+      last_released_tag="$(snap info $SNAPCRAFT_PROJECT_NAME | awk '$1 == "beta:" { print $2 }')"
+      # If the latest tag from the upstream project has not been released to
+      # beta, build that tag instead of master.
+      if [ "${last_committed_tag_ver}" != "${last_released_tag}" ]; then
+        git fetch
+        git checkout "${last_committed_tag}"
+      fi
+      snapcraftctl set-version "$(git describe --tags | sed 's/v//')"
+      override-build: |
```
Comment (ibuildthecloud) [inline snap/snapcraft.yaml]: is this supposed to be indented less?  Also, can we build using our build scripts?

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      export GOPATH=$PWD
+      cd src/github.com/rancher/k3s
+      env CGO_ENABLED=0 GOOS=linux \
+      go build --ldflags "-s -w \
+        -X 'github.com/rancher/k3s/version.GitCommit=$(git rev-list -1 HEAD)' \
+        -X 'github.com/rancher/k3s/version.Version=$(git describe --tags --abbrev=0)'" \
+        -a -installsuffix cgo -o $SNAPCRAFT_PART_INSTALL/bin/k3s
```
Comment (ibuildthecloud) [inline snap/snapcraft.yaml]: I think this is missing containerd.  Does this fully work?

File: snap/snapcraft.yaml (hunk 1 above)
Comment (mikeroyal) [reply snap/snapcraft.yaml]: Yes, you could use your own build scripts.

File: snap/snapcraft.yaml (hunk 2 above)
Comment (mikeroyal) [reply snap/snapcraft.yaml]: I need to update it

**Comment by cjellick** [comment]
Is this still relevant? Even though it's old, seems like it could be.
I'm going to close due to age, but feel free to reopen if you think it's still relevant. Thanks!

**Pull request closed without merging**